odoo/addons/base/models/ir_fields.py

```diff
@@ -203,8 +210,80 @@ def _str_to_json(self, model, field, value):
+
+        warnings = []
+        for property_value in value:
+            if {'name', 'type', 'string'} > property_value.keys():
+                msg = _("'%(value)s' does not seem to be a valid Properties value for field '%%(field)s'. A Properties should contain a list of property. Each property need at least 'name', 'type' and 'string' attribute")
+                raise self._format_import_error(ValueError, msg, {'value': property_value})
+
+            val = property_value.get('value')
+            if not val:
+                continue
+
+            if property_value['type'] == 'selection':
+                # either label or the technical value
+                new_val = next(iter(
+                    sel_val for sel_val, sel_label in property_value['selection']
+                    if val in (sel_val, sel_label)
+                ), None)
+                if not new_val:
+                    msg = _("'%(value)s' does not seem to be a valid Selection value for '%(description_selection)s' (of '%%(field)s' Properties field).")
+                    raise self._format_import_error(ValueError, msg, {'value': val, 'description_selection': property_value['string']})
+                property_value['value'] = new_val
+
+            elif property_value['type'] == 'tags':
+                tags = val.split(',')
+                new_val = []
+                for tag in tags:
+                    val_tag = next(iter(
+                        tag_val for tag_val, tag_label, _color in property_value['tags']
+                        if tag in (tag_val, tag_label)
+                    ), None)
+                    if not val_tag:
+                        msg = _("'%(value)s' does not seem to be a valid Tag value for '%(description_tag)s' (of '%%(field)s' Properties field).")
+                        raise self._format_import_error(ValueError, msg, {'value': tag, 'description_selection': property_value['string']})
+                    new_val.append(val_tag)
+                property_value['value'] = new_val
+
+            elif property_value['type'] == 'boolean':
+                new_val, warnings = self._str_to_boolean(model, field, val)
+                if not warnings:
+                    property_value['value'] = new_val
+                else:
+                    msg = _("Unknown value '%(value)s' for boolean '%(desc_property)s' property (of '%%(field)s' Properties field).")
+                    raise self._format_import_error(ValueError, msg, {'value': val, 'description_selection': property_value['string']})
+
+            elif property_value['type'] in ('many2one', 'many2many'):
+                [record] = property_value['value']
+
+                subfield, w1 = self._referencing_subfield(record)
+                if w1:
+                    warnings.append(w1)
+
+                values = record[subfield]
+
+                references = values.split(',') if property_value['type'] == 'many2many' else [values]
```

Comment:
You could always split and test that for many2one you have at most 1 value.

Reply:
Nop I don't want conflict with potential value with ',' for many2one.